.github/workflows/push-go-runtime.yml

```diff
@@ -0,0 +1,72 @@
+# Workflows pertaining to the jsii/superchain Docker image
+name: Push Jsii Runtime Go
+
+on:
+  push:
+    branches: [main]
+
+jobs:
+  push-go-runtime:
+    name: Install Dependencies
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out
+        uses: actions/checkout@v2
+      - name: Set up Node 10
+        uses: actions/setup-node@v2.1.2
+        with:
+          node-version: '10'
+      - uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+      - name: Install Dependencies
+        run: |-
+          # TypeScript project dependencies
+          yarn install --frozen-lockfile
+      - name: Align Versions
+        run: |-
+          ./scripts/align-version.sh
+      - name: Build dependencies
+        run: |-
+          yarn build \
+          --scope codemaker \
+          --scope @jsii/spec \
+          --scope @jsii/kernel \
+          --scope @jsii/runtime
+      - name: Build
+        run: |-
+          yarn build \
+          --scope @jsii/go-runtime
+      - name: Build Repo Directory
+        id: build-repo-dir
+        working-directory: $GITHUB_WORKSPACE/packages/@jsii/go-runtime
+        env:
+          JSII_RUNTIME_REPO_NAME: ${{ secrets.JSII_RUNTIME_REPO_NAME }}
+          JSII_RUNTIME_REPO_USERNAME: ${{ secrets.JSII_RUNTIME_REPO_USERNAME }}
+          JSII_RUNTIME_REPO_TOKEN: ${{ secrets.AUTO_APPROVE_GITHUB_TOKEN }}
+        run: |-
+          REPO_DIR=$(mktemp -d)
+          npx ts-node ./build-tools/build-repo-dir.ts $REPO_DIR
+          echo "::set-output name=repodir::$REPO_DIR"
+      - name: Check for changes
+        id: check-for-changes
+        run: |-
+          # Change directory manually as using `working-directory` doesn't seem to support using outputs from the previous step
+          cd ${{ steps.build-repo-dir.outputs.repodir }}
+          git fetch --depth=1 --quiet origin main
+          git add .
+          # Check for modifications in tracked files
+          changed=$(git diff --name-only origin/main)
+          if [-z "$changed"]; then
+            echo "::set-output name=result::false"
+          else
+            echo "::set-output name=result::true"
+          fi
+      - name: Commit and Push
+        if: steps.check-for-changes.outputs.result == 'true'
```

Comment:
@RomainMuller when testing locally using `act`, I can't get this step to run even when verifying that the output is correctly set in the previous step. Wondering if you have some insight on this?

Reply:
Could be a bug in `act`? I guess the question is does it behave "as intended" on GitHub Actions? If it does, then maybe report to `nektos/act`?

Reply:
Additionally... You should be able to make the "Push" unconditional (worst case, result in "already up to date" idempotent success).

Basically, in the "detect" step you'd check for changes and create the commit as appropriate... in the second step you simply (always) push.

Reply:
I kept this for now and assumed its a bug with act. If running in actions doesn't work I'll remove and just always push.